src/components/PostsList.tsx

```diff
@@ -1,54 +1,62 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
 
 import classNames from 'classnames';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Post } from '../types/Post';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { selectPost } from '../features/CurrentPostAPI/CurrentPostAPI';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+export const PostsList: React.FC = () => {
+  const { posts } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.currentPost.post);
+  const dispatch = useAppDispatch();
+  const selectedUser = useAppSelector(state => state.author.user);
+
+  const handleSelectedPost = (post: Post) => {
+    dispatch(selectPost({ post }));
+  };
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+  useEffect(() => {
+    dispatch(selectPost({ post: null }));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedUser]);
 
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          <th> </th>
-        </tr>
-      </thead>
+  return (
+    <div data-cy="PostsList">
+      <p className="title">Posts:</p>
 
-      <tbody>
-        {posts.map(post => (
-          <tr key={post.id} data-cy="Post">
-            <td data-cy="PostId">{post.id}</td>
-            <td data-cy="PostTitle">{post.title}</td>
-            <td className="has-text-right is-vcentered">
-              <button
-                type="button"
-                data-cy="PostButton"
-                className={classNames('button', 'is-link', {
-                  'is-light': post.id !== selectedPostId,
-                })}
-                onClick={() => {
-                  onPostSelected(post.id === selectedPostId ? null : post);
-                }}
-              >
-                {post.id === selectedPostId ? 'Close' : 'Open'}
-              </button>
-            </td>
+      <table className="table is-fullwidth is-striped is-hoverable is-narrow">
+        <thead>
+          <tr className="has-background-link-light">
+            <th>#</th>
+            <th>Title</th>
+            <th> </th>
```

Comment:
The empty <th> element on this line might be intended for spacing or alignment purposes, but it could be beneficial to add a label or remove it if it's not necessary for accessibility reasons.